src/utils.js

```diff
@@ -232,8 +287,96 @@ var nmPhoneUtils = {
    * @param  {Number} length
    * @return {String}
    */
-  shortenToLength: function (value, length) {
-    return value.substring(0, length);
+  shortenToLength: shortenToLength,
+
+  /**
+   * Format the phoneData object into a string of the form:
+   *
+   * +<countrycode>-[<carriercode>-]<number>
+   *
+   * @param  {Object} phoneData
+   * @return {String}
+   */
+  getFormattedNumber: function getFormattedNumber(phoneData) {
+    if (!phoneData.cellTokens) {
+      return '';
+    }
+
+    return [
+        (!!phoneData.cellTokens.countryCode ? '+' + phoneData.cellTokens.countryCode : ''),
+        (''+phoneData.cellTokens.carrierCode),
+        (''+phoneData.cellTokens.number),
+      ].filter(function(v) { return !_.isEmpty(v); }).join('-');
+  },
+
+  isPhoneDataValid: function isPhoneDataValid(phoneData, phoneSettings, min, max) {
+    return (
+      !!phoneData.fkCountry &&
+      this.validateCountry(phoneData.fkCountry, phoneSettings).valid &&
+      this.validateCarrierCode(phoneData.cellTokens.carrierCode, phoneData.fkCountry, phoneSettings).valid &&
+      !!phoneData.cellTokens.number &&
+      this.validateNumber(phoneData.cellTokens.number, min, max).valid
+    );
+  },
+
+  /**
+   * Return the numeric validator function.
+   * The numericValidator function will filter out non numeric characters from the given input and call the callback.
+   * The callback is invoked only if one or more characters are removed from the input string
+   *
+   * @param callback
+   * @returns {*}
+   */
+  getNumericValidator: function getNumericValidator(callback) {
+    return function numericValidator(input) {
+      var output;
+
+      if (_.isString(input)) {
+        output = extractNumbers(input);
+
+        if (input !== output) {
+          callback(output);
+        }
+      }
+
+      return output;
+    };
+  },
+
+  /**
+   * Parser and formatter to invalidate a field which its value length exceeded the maximum number of
+   * allowed characters
+   *
+   * @param value
+   * @returns {*}
+   */
+  getMaxLengthValidator: function getMaxLengthValidator(callback, maxlength) {
+    return function maxLengthValidator(value) {
+      var isValid = _.isEmpty(value) || value.length <= maxlength;
+      if (!(isValid || _.isEmpty(value))) {
+        value = value.substring(0, maxlength);
+      }
+
+      callback(isValid, value);
+
+      return value;
+    }
+  },
+
+  /**
+   * Parser and Formatter to invalidate a field which its value length goes below the minimum number of
+   * allowed characters
+   *
+   * @param value
+   * @returns {*}
+   */
+  getMinLengthValidator: function getMinLengthValidator(callback, minlength) {
+    return function minLengthValidator(value) {
```

Comment:
@Mohamed-amin In this case I think I just didn't follow the convention :-D

Moreover the `minLengthValidator` function will be passed to the angular [controller's `$parsers` / `$formatters`](https://github.com/namshi/nmPhone/blob/directive-as-part-of-a-form-downgrade-lodash/src/nmRangeLength.js#L52) and I think it will be called with only one `value` parameter.

@shidhincr I'll update the order of the parameters.